### PR TITLE
PowerBI and Jira Search connectors

### PR DESCRIPTION
**Pull request by paulculmsee**
First pull request ever - be gentle :-)

---
When submitting a connector, please check the following conditions for your PR.

- [x] I attest that the connector works and I verified by deploying and testing all the operations.
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`.

### CODE REVIEW

**Comment by Laskewitz** [comment]
Well done @paulculmsee! Congrats on the first PR! 🥳

File: independent-publisher-connectors/AzureADExtensions/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,504 @@
+{
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AzureADExtensions/apiDefinition.swagger.json:1]: @paulculmsee Please let us know the reason to add `AzureAdExtensions` connector files where the PR title is for `PowerBI` and `Jira` conectors.

File: independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,392 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "JIRA Search",
+    "description": "Searches for JIRA issues using JQL expression using the GET pattern. More information can be found at: https://docs.atlassian.com/software/jira/docs/api/REST/8.16.2/#api/2/search-search",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json:5]: @paulculmsee The description value can be improved by adding some more information about to your connector product or service.

Documentation - [Description](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#description)

File: independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+                },
+                "startAt": {
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "startAt"
+                },
+                "maxResults": {
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "maxResults"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json:41]: @paulculmsee Can you please provide valid description values of the properties as they seem like property Ids.

Documentation - [Description values of properties](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#operation-and-parameter-summaries-and-descriptions)

Applies to all the similar description values in the swagger.

File: independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+            "description": "The maximum number of issues to return (defaults to 50). The maximum allowable value is dictated by the Jira property 'jira.search.views.default.max'. If you specify a value that is higher than this number, your search results will be truncated.",
+            "x-ms-summary": "Max results to return",
+            "default": 50
+          }
+        ]
+      }
+    }
+  },
+  "x-ms-connector-metadata" : [
+	  {
```
Comment (sriyen-msft) [inline independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json:365]: @paulculmsee Please fix indentation issues in this property.

File: independent-publisher-connectors/JiraSearch/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      {
+        "templateId": "dynamichosturl",
+        "title": "Apply Host URL",
+        "parameters": {
+          "x-ms-apimTemplateParameter.urlTemplate": "https://@queryParameters('hostname').atlassian.net/"
+        }
+      }
+    ],
+    "publisher": "Paul Culmsee",
+	"stackOwner" : "Atlassian"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/JiraSearch/apiProperties.json:43]: @paulculmsee Please remove the space to fix indentation for `stackOwner` property.

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1062 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Default title",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:4]: @paulculmsee Can you please provide valid title for the connector ?

Documentation - [Title of connector](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#title)

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1062 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Default title",
+    "description": "Custom connector that connects to the Power BI Admin API (https://docs.microsoft.com/en-us/rest/api/power-bi/admin)",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:5]: @paulculmsee Can you please provide valid description value for connector ?

Documentation - [Description of connector](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#description)

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1062 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Default title",
+    "description": "Custom connector that connects to the Power BI Admin API (https://docs.microsoft.com/en-us/rest/api/power-bi/admin)",
+    "version": "1.0",
+	"contact" : {
+		"name" : "Paul Culmsee",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:8]: @paulculmsee Please fix all indentation issues in the swagger

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,1062 @@
...
+                        "type": "string",
+                        "description": "id"
+                      },
+                      "isReadOnly": {
+                        "type": "boolean",
+                        "description": "isReadOnly"
+                      },
+                      "isOnDedicatedCapacity": {
+                        "type": "boolean",
+                        "description": "isOnDedicatedCapacity"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:53]: @paulculmsee Can you please provide valid description values for all the properties in swagger ?

Documentation - [Description values](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#operation-and-parameter-summaries-and-descriptions)

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,1062 @@
...
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+			"x-ms-summary" : "Number of records to skip",
+            "description": "Use the top parameter to skip the first x amount of groups that are being returned."
+          }
+        ]
+      }
+    },
+    "/v1.0/myorg/admin/groups/{groupId}/users": {},
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:122]: @paulculmsee Please remove all the empty operations as they will not add any value to the connector.

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,1062 @@
...
+                      }
+                    }
+                  },
+                  "description": "value"
+                }
+              }
+            }
+          }
+        },
+        "summary": "DatasetsGetDataSourcesAsAdmin",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:963]: @paulculmsee Documentation - [Summary Values](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#operation-and-parameter-summaries-and-descriptions)

File: independent-publisher-connectors/PowerBI/apiProperties.json [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+            "required": "false",
+            "hidden": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
+    "capabilities": [],
+    "publisher": "Paul Culmsee"
+
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiProperties.json:47]: @paulculmsee Please provide `stackOwner` property as well

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AzureADExtensions/apiDefinition.swagger.json (hunk 1 above)
Comment (paulculmsee) [reply independent-publisher-connectors/AzureADExtensions/apiDefinition.swagger.json:1]: My mistake - still learning and will remove

File: independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json (hunk 2 above)
Comment (paulculmsee) [reply independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json:5]: Added more detail

File: independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json (hunk 3 above)
Comment (paulculmsee) [reply independent-publisher-connectors/JiraSearch/apiDefinition.swagger.json]: Applied to all JIRA's publicly disclosed issue field types as per their documentation. Could not do them all however

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json (hunk 9 above)
Comment (paulculmsee) [reply independent-publisher-connectors/PowerBI/apiDefinition.swagger.json]: done

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json (hunk 10 above)
Comment (paulculmsee) [reply independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:122]: done

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,1058 @@
...
+                        "type": "string",
+                        "description": "id"
+                      },
+                      "name": {
+                        "type": "string",
+                        "description": "name"
+                      },
+                      "addRowsAPIEnabled": {
+                        "type": "boolean",
+                        "description": "addRowsAPIEnabled"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:826]: @paulculmsee Can you please provide valid description values, there are few other description properties need valid values.

File: independent-publisher-connectors/PowerBI/apiDefinition.swagger.json (hunk 11 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/PowerBI/apiDefinition.swagger.json:959]: @paulculmsee Can you please fix it

File: independent-publisher-connectors/PowerBI/apiProperties.json (hunk 12 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/PowerBI/apiProperties.json:47]: @paulculmsee The `stackOwner` property is not added.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by paulculmsee** [comment]
Will resubmit as 2 separate connectors

**Pull request closed without merging**